google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/ChannelCache.java

```diff
@@ -20,14 +20,24 @@
 import io.grpc.Channel;
 import io.grpc.ManagedChannel;
 import io.grpc.ManagedChannelBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.TimeUnit;
 import java.util.function.Function;
 
 /** A ChannelCache creates and stores default channels for use with api methods. */
 public class ChannelCache {
   private final Function<String, ManagedChannel> channelFactory;
-  private final ConcurrentHashMap<String, ManagedChannel> channels = new ConcurrentHashMap<>();
+  private final ConcurrentHashMap<String, List<ManagedChannel>> channels =
+      new ConcurrentHashMap<>();
+
+  private static final Random rand = new Random();
```

Comment:
Don't do random, do round robin. Just store a Queue<ManagedChannel> for each, pull off and push onto the back, and use synchronized instead of ConcurrentHashMap to ensure consistency.